src/test/java/com/klodnicki/Bike/v2/service/RentBikeServiceHandlerTest.java

```diff
@@ -0,0 +1,263 @@
+package com.klodnicki.Bike.v2.service;
+
+import com.klodnicki.Bike.v2.DTO.bike.BikeForNormalUserResponseDTO;
+import com.klodnicki.Bike.v2.DTO.rent.RentRequestDTO;
+import com.klodnicki.Bike.v2.DTO.rent.RentResponseDTO;
+import com.klodnicki.Bike.v2.model.RentRequest;
+import com.klodnicki.Bike.v2.model.entity.Bike;
+import com.klodnicki.Bike.v2.model.entity.ChargingStation;
+import com.klodnicki.Bike.v2.model.entity.Rent;
+import com.klodnicki.Bike.v2.model.entity.User;
+import com.klodnicki.Bike.v2.repository.BikeRepository;
+import com.klodnicki.Bike.v2.repository.ChargingStationRepository;
+import com.klodnicki.Bike.v2.repository.RentRepository;
+import com.klodnicki.Bike.v2.repository.UserRepository;
+import com.klodnicki.Bike.v2.service.api.ChargingStationServiceApi;
+import com.klodnicki.Bike.v2.service.api.UserServiceApi;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.modelmapper.ModelMapper;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class RentBikeServiceHandlerTest {
+
+    private RentRepository rentRepository;
+    private BikeRepository bikeRepository;
+    private ChargingStationRepository chargingStationRepository;
+    private UserRepository userRepository;
+    private ModelMapper modelMapper;
+    private RentBikeServiceHandler rentBikeServiceHandler;
+    private BikeServiceHandler bikeServiceHandler;
+    private ChargingStationServiceApi chargingStationService;
+    private UserServiceApi userService;
+    private ChargingStation chargingStation;
+    private Rent rent;
+    private User user;
+    private Bike bike;
+    private List<Bike> bikeList;
```

Comment:
I see no reason for those fields to actually be fields.
The other fields – like services for example – are fine as you're mocking them and then in some methods need to configure them with `when()` method.

Also, as I'm talking about mocking – I don't feel like there is a need for mocks of those instances. There is no logic in those classes and inside of them methods that could influence the result of the code. To be honest I'm not sure how to treat it as this solution will work and is readable in some way I guess, but I feel like I'm not used to tests written like that. If I were to write those I'd not make fields out of these and wouldn't mock them – just as you don't mock DTO classes you're using below. I think this would make code cleaner actually and easier – for me at least ;) – to read.

I'd like to know what you think and if this code you think is readable. If so, I can approve it. Otherwise you can change it to make it more into my vision, but surely it is not a necessity.

Reply:
Oh sorry, yes you are right! I will make some corrections!

Reply:
Great! I'm looking forward to it.

Reply:
Okay! check it out now! :)